### DISCUSSION TITLE
sql formatting

### DISCUSSION THREAD

**Discussion by JeroenNienhuis** [dbeaver/dbeaver#13497] [Q&A] [answered]
Often I have a (very) long sql strings without line breaks (e.g. generated dynamically). There is no problem running the code, but when I want to understand what is written or try to improve the result one long single line is not helpful. The solution is formatting the string by breaking it up in lines and adding indents. In Notepad++ (Windows) I use the Poor manś T-SQL formatter plugin (http://architectshack.com/poormanstsqlformatter.ashx) to do this, but I'm not always working under Windows.

Would it be possible to add some kind of sql formatting routine to make long sql lines better readable?

**Comment by p91paul** [comment, marked as answer]
Try ctrl+shift+f. It works on the whole script, but if you select some text formatting is applied to the selection only.
There is some configuration in preferences for this feature, and you can also use an external formatter.
